mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/api/akka/route/proto/JobClusterProtoAdapter.java

```diff
@@ -77,7 +79,7 @@ public static final CreateJobClusterRequest toCreateJobClusterRequest(final Name
                 jd.getMigrationConfig(),
                 jd.getIsReadyForJobMaster(),
                 jd.getParameters(),
-                processLabels(jd)
+                processLabels(jd, defaultResourceClusterId)
```

Comment:
Can we change the NamedJobDefinition to include the resourceClusterId? It would make the code look cleaner.

Reply:
this change is to override with a default to be back compact with current API behavior so anyone using existing API will not fall back to mesos stack. Since we are already having a internal "_mantis.resourceCluster" label being used I guess what you propose is more like a v2 API for job cluster APIs we can follow up with?

Reply:
I don't think it's necessary to change the way data is stored within a JobDefinition to access this property. Instead, I suggest adding a getResourceCluster() function to the JobDefinition class that always returns a valid resource cluster ID. This way, we avoid duplicating the logic in multiple locations. Eventually, we can update the data structure to store the property in a different manner, as you suggested.